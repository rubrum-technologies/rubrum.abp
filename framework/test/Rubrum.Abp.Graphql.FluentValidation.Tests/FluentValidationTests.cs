using CookieCrumble;
using Shouldly;
using Xunit;

namespace Rubrum.Abp.Graphql;

public class FluentValidationTests : GraphqlFluentValidationTestBase
{
    [Fact]
    public async Task Successfully()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            """
            mutation {
                createCountry(input: { name: "Test" }) {
                    country {
                        name
                    }
                    errors {
                        ... on Error {
                            message
                        }
                    }
                }
            }
            """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }

    [Fact]
    public async Task Fail()
    {
        await using var result = await ExecuteRequestAsync(b => b.SetQuery(
            """
            mutation {
                createCountry(input: { name: "TestTestTestTestTestTest" }) {
                    country {
                        name
                    }
                    errors {
                        ... on Error {
                            message
                        }
                    }
                }
            }
            """));

        result.ShouldNotBeNull();

        result.MatchSnapshot();
    }
}
