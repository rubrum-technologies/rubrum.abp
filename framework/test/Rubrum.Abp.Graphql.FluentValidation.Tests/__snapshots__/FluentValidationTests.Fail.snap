{
  "data": {
    "createCountry": {
      "country": null,
      "errors": [
        {
          "message": "There are 1 validation errors:\r\nThe length of \u0027Maximum length 20\u0027 must be 20 characters or fewer. You entered 24 characters. (Name)\r\n"
        }
      ]
    }
  }
}